### PR TITLE
drivers: interrupt_controller: plic: support edge triggered interrupts

### PR DESCRIPTION
**Pull request by jgl-meta**
This adds a check and option for edge triggered interrupts

### CODE REVIEW

**Comment by jgl-meta** [comment]
@cfriedt can you take a look at this when you have a chance?

File: drivers/interrupt_controller/intc_plic.c [hunk 1]
```diff
@@ -29,13 +29,39 @@
...
+ * value of the irq. In the event edge irq is not supported this
+ * routine will return 0
+ *
+ * @param irq IRQ number to add to the trigger
+ *
+ * @return irq value when enabled 0 otherwise
+ */
+static int riscv_plic_is_edge_irq(uint32_t irq)
+{
+#if defined(CONFIG_PLIC_SUPPORTS_EDGE_IRQ)
```
Comment (cfriedt) [inline drivers/interrupt_controller/intc_plic.c:55]: Could this be a `if (IS_ENABLED(..))` or even `if (!IS_ENABLED(..))`?

File: drivers/interrupt_controller/intc_plic.c [hunk 2]
```diff
@@ -29,13 +29,39 @@
 #define PLIC_IRQS        (CONFIG_NUM_IRQS - CONFIG_2ND_LVL_ISR_TBL_OFFSET)
 #define PLIC_EN_SIZE     ((PLIC_IRQS >> 5) + 1)
 
+#define EDGE_TRIG_TYPE   (PLIC_MAX_PRIO + 0x1080)
+#define EDGE_TRIG_SHIFT  5
```
Comment (ycsin) [inline drivers/interrupt_controller/intc_plic.c:33]: I would probably namespace these with `PLIC_*`

File: drivers/interrupt_controller/intc_plic.c [hunk 3]
```diff
@@ -29,13 +29,39 @@
 #define PLIC_IRQS        (CONFIG_NUM_IRQS - CONFIG_2ND_LVL_ISR_TBL_OFFSET)
 #define PLIC_EN_SIZE     ((PLIC_IRQS >> 5) + 1)
 
+#define EDGE_TRIG_TYPE   (PLIC_MAX_PRIO + 0x1080)
```
Comment (ycsin) [inline drivers/interrupt_controller/intc_plic.c:32]: and maybe some reference for the `0x1080`

**Review by ycsin — COMMENTED** [review]
just some nits

File: drivers/interrupt_controller/Kconfig.plic [hunk 4]
```diff
@@ -10,3 +10,10 @@ config PLIC
 	help
 	  Platform Level Interrupt Controller provides support
 	  for external interrupt lines defined by the RISC-V SoC.
+
+config PLIC_SUPPORTS_EDGE_IRQ
+	bool "The given interrupt controller supports edge interrupts"
+	default n
```
Comment (ycsin) [inline drivers/interrupt_controller/Kconfig.plic:16]:
```suggestion
```

This line is probably redundant

File: drivers/interrupt_controller/intc_plic.c (hunk 3 above)
Comment (cfriedt) [reply drivers/interrupt_controller/intc_plic.c]: Yeah, I was also curious about that offset.

File: drivers/interrupt_controller/intc_plic.c (hunk 1 above)
Comment (ycsin) [reply drivers/interrupt_controller/intc_plic.c]: Do you mean avoid the use of preprocessor magic?

**Comment by dcpleung** [comment]
Suggestion: change the commit summary prefix to `interrupt_controller: plic:` to make it clear that this is for the PLIC driver. Currently it sounds like this adds edge triggered interrupts for all drivers.

**Comment by dcpleung** [comment]
Could you also update the commit message to indicate it is for plic? That would help when looking at git log (especially with oneline).

**Review by ycsin — APPROVED** [review]
LGTM

**Review by cfriedt — APPROVED** [review]

**Pull request merged**